### PR TITLE
test: some branches in view per diem page

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=-bAq6uF)
### <samp>🤖 Generated by Copilot at 5774a73</samp>

This pull request enhances the `ViewPerDiemPage` component by improving its code style, readability, and test coverage. It removes unnecessary commas and uses the `includes` method for array checks in `view-per-diem.page.ts`. It also adds new tests for the `ionViewWillEnter` method and its dependencies in `view-per-diem.page.spec.ts`.

### <samp>🤖 Generated by Copilot at 5774a73</samp>

> _Oh we're the coders of the sea, and we work with skill and glee_
> _We tidy up our code and make it shine_
> _We remove the extra commas from our `ViewPerDiemPage`_
> _And we test our `ionViewWillEnter` fine_

### <samp>🤖 Generated by Copilot at 5774a73</samp>

*  Remove unnecessary commas at the end of function calls for code style consistency and readability ([link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057L204-R204), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057L214-R214), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057L300-R300), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057L394-R443), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L112-R112), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L175-R178), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L188-R195), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L203-R211), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L235-R238), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L253-R252), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L259-R259), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L267-R266), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L284-R281), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L291-R288), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L306-R303), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L312-R309), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L387-R389))
*  Replace `indexOf` with `includes` for checking array membership for code readability and conciseness ([link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-b66c53e7e9d2eaf1447540624a0e92ad7ea25c5299edf4e5720110afb75cec90L274-R275))
*  Add test cases for `ionViewWillEnter` method of `ViewPerDiemPage` class to increase test coverage and ensure correctness of logic ([link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057R330-R371), [link](https://github.com/fylein/fyle-mobile-app/pull/2524/files?diff=unified&w=0#diff-aa7434f7715887f5188bf634482b759f816a20bd7f5d51ef219f9754ec71a057R418-R424))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/view-per-diem/view-per-diem.page.ts [hunk 1]
```diff
@@ -232,11 +231,11 @@ export class ViewPerDiemPage {
     forkJoin([this.txnFields$, this.extendedPerDiem$.pipe(take(1))])
       .pipe(
         map(([expenseFieldsMap, extendedPerDiem]) => {
-          this.projectFieldName = expenseFieldsMap?.project_id && expenseFieldsMap?.project_id[0]?.field_name;
-          const isProjectMandatory = expenseFieldsMap?.project_id && expenseFieldsMap?.project_id[0]?.is_mandatory;
+          this.projectFieldName = expenseFieldsMap?.project_id && expenseFieldsMap.project_id[0]?.field_name;
```
Comment (suyashpatil78) [inline src/app/fyle/view-per-diem/view-per-diem.page.ts:234]: Change here
Second `?.` was not needed here

**Comment by suyashpatil78** [comment]
rest all are linting changes

File: src/app/fyle/view-per-diem/view-per-diem.page.ts [hunk 2]
```diff
@@ -273,22 +272,21 @@ export class ViewPerDiemPage {
       filter(() => this.view === ExpenseView.team),
       map(
         (etxn) =>
-          ['COMPLETE', 'POLICY_APPROVED', 'APPROVER_PENDING', 'APPROVED', 'PAYMENT_PENDING'].indexOf(etxn.tx_state) >
-          -1,
-      ),
+          ['COMPLETE', 'POLICY_APPROVED', 'APPROVER_PENDING', 'APPROVED', 'PAYMENT_PENDING'].indexOf(etxn.tx_state) > -1
+      )
```
Comment (Chethan-Fyle) [inline src/app/fyle/view-per-diem/view-per-diem.page.ts:276]: `['COMPLETE', 'POLICY_APPROVED', 'APPROVER_PENDING', 'APPROVED', 'PAYMENT_PENDING'].includes(etxn.tx_state)`

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/fyle/view-per-diem/view-per-diem.page.ts [hunk 3]
```diff
@@ -270,9 +270,8 @@ export class ViewPerDiemPage {
 
     this.canFlagOrUnflag$ = this.extendedPerDiem$.pipe(
       filter(() => this.view === ExpenseView.team),
-      map(
-        (etxn) =>
-          ['COMPLETE', 'POLICY_APPROVED', 'APPROVER_PENDING', 'APPROVED', 'PAYMENT_PENDING'].indexOf(etxn.tx_state) > -1
+      map((etxn) =>
+        ['COMPLETE', 'POLICY_APPROVED', 'APPROVER_PENDING', 'APPROVED', 'PAYMENT_PENDING'].includes(etxn.tx_state)
```
Comment (suyashpatil78) [inline src/app/fyle/view-per-diem/view-per-diem.page.ts:274]: Used `includes` instead of `indexOf` as per PR comment

**Pull request merged**